cpu/cortexm_common/vectors_cortexm.c

```diff
@@ -110,6 +111,52 @@ void nmi_default(void)
 
 void hard_fault_default(void)
```

Comment:
> There is a risk of corrupting the stack pointer that we need to use in the asm block below if this function is not naked, because some extra stacking may be added by the compiler.

Thats right. I already asked myself why the handler is not naked, because it won't return anyway.